samples/matter/lock/src/app_task.h

```diff
@@ -53,7 +53,14 @@ class AppTask {
 
 	friend AppTask &GetAppTask();
 
-	enum class TimerFunction { NoneSelected = 0, SoftwareUpdate, FactoryReset };
+	enum class TimerFunction {
+		NoneSelected = 0,
+		SoftwareUpdate,
+		FactoryReset,
+#if CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP
```

Comment:
I guess this is fine. What if we add `buttons` to .dts of nRF7002dk board and instead of doing #if BOARD, could we instead check the number of supported buttons on the board?

Reply:
Do you have any hints how to do that? I took a look on ARRAY_SIZE(buttons) that is used in the dk_leds_and_buttons, but it doesn't seem such comparison can be done in the preprocessor, as array needs to be declared.

Reply:
I would prefer using preprocessor only solution to don't keep unused code.

Reply:
@kkasperczyk-no what about sth like this:

https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/led_pwm/src/main.c#L17-L25

Or maybe something like this :)?

```suggestion
#define INCREMENT_BY_ONE +1
#define NUMBER_OF_LEDS (0 DT_FOREACH_CHILD(LEDS_NODE_ID, COUNT_LED))
```

Reply:
Almost correct, it should be `#define INCREMENT_BY_ONE(some_ignored_arg) +1`, nevertheless it's a great idea!